Scripts/PAIF-N/02-install-nvidia-driver-vlcm.ps1

```diff
@@ -0,0 +1,186 @@
+# Broadcom Inc. and/or its subsidiaries.
+#>
+
+<#
+.SYNOPSIS
+
+This script configures the ESXi host for AI workloads
+
+.DESCRIPTION
+
+This script configures the ESXi host for AI workloads which includes installing the
+NVIDIA AI Enterprise vGPU driver and NVIDIA GPU Management Daemon on the ESXi hosts.
+vLCM is used for that purpose.
+
+The script changes the default graphics type of the GPU devices to Shared Direct. The Xorg
+service is then restarted. Finally, the vLCM is used to install the NVIDIA GPU driver and
+management daemon.
+
+.NOTES
+
+Prerequisites:
+ - VI workload domain (vCenter server instance)
+ - ESXi hosts with GPUs
+
+"Global parameters", "Workload domain parameters", "GPU parameters" should be updated to
+reflect the environment they are run in. This may require altering the spec creation script.
+
+#>
+
+$ErrorActionPreference = 'Stop'
+
+# --------------------------------------------------------------------------------------------------------------------------
+# Global parameters
+# --------------------------------------------------------------------------------------------------------------------------
+
+$domainName = 'sfo-w01'
+$domain = 'vrack.vsphere.local'
+
+# --------------------------------------------------------------------------------------------------------------------------
+# Workload domain parameters - stripped down version of $domainSpec from 01-deploy-vcf-workload-domain.ps1
+
+$domainSpec = @{
+   VCenterSpec = @{
+      RootPassword = "VMware123!"
+      NetworkDetailsSpec = @{
+         DnsName = "$DomainName-vc01.$domain"
+      }
+   }
+   ComputeSpec = @{
+      ClusterSpecs = @(
+         @{
+            Name = "$DomainName-cl01"
+         }
+      )
+   }
+}
+# --------------------------------------------------------------------------------------------------------------------------
+# GPU parameters
+
+$nvidiaDriverLocation = "http://NVIDIA-VGPU-DRIVER-LOCATION/"
+$gpuParameters = @{
+   EsxiImageName = "8.0 U2b - 23305546"
+   NVIDIA = @(
+      @{
+         Location = "$nvidiaDriverLocation/NVD-AIE-800_550.54.16-1OEM.800.1.0.20613240_23471877.zip"
+         Name = "NVIDIA AI Enterprise vGPU driver for VMWare ESX-8.0.0"
+         Version = "550.54.16"
+         Description = 'NVIDIA AI Enterprise vGPU driver for VMWare ESX-8.0.0'
+      },
+      @{
+         Location = "$nvidiaDriverLocation/nvd-gpu-mgmt-daemon_550.54.16-0.0.0000_23475823.zip"
+         Name = "NVIDIA GPU monitoring and management daemon"
+         Version = "550.54.16 - Build 0000"
+         Description = "NVIDIA GPU monitoring and management daemon"
+      }
+   )
+   GraphicsType = 'sharedDirect'
+   HostDefaultGraphicsType = 'sharedDirect'
+   SharedPassthruAssignmentPolicy = 'performance'
+}
+# --------------------------------------------------------------------------------------------------------------------------
+
+# Connect to the VC of the workload domain
+$vcConn = Connect-VIServer `
```

Comment:
Maybe have it as -NotDefault and pass it to the corresponding commands, in case this is run a session where there are other connections?

Reply:
This will require major refactoring. Will consider it for the next pass.